src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -255,4 +257,114 @@ QgsPointCloudLayerChunkedEntity::~QgsPointCloudLayerChunkedEntity()
   cancelActiveJobs();
 }
 
+QVector<QgsRayCastingUtils::RayHit> QgsPointCloudLayerChunkedEntity::rayIntersection( const QgsRayCastingUtils::Ray3D &ray, const QgsRayCastingUtils::RayCastContext &context ) const
+{
+  QVector<QgsRayCastingUtils::RayHit> result;
+  QgsPointCloudLayerChunkLoaderFactory *factory = static_cast<QgsPointCloudLayerChunkLoaderFactory *>( mChunkLoaderFactory );
+
+  // transform ray
+  const QgsVector3D originMapCoords = factory->mMap.worldToMapCoordinates( ray.origin() );
+  const QgsVector3D pointMapCoords = factory->mMap.worldToMapCoordinates( ray.origin() + ray.origin().length() * ray.direction().normalized() );
+  QgsVector3D directionMapCoords = pointMapCoords - originMapCoords;
+  directionMapCoords.normalize();
+
+  const QVector3D rayOriginMapCoords( originMapCoords.x(), originMapCoords.y(), originMapCoords.z() );
+  const QVector3D rayDirectionMapCoords( directionMapCoords.x(), directionMapCoords.y(), directionMapCoords.z() );
+
+  const int screenSizePx = std::max( context.screenSize.height(), context.screenSize.width() );
+
+  const QgsPointCloud3DSymbol *symbol = factory->mSymbol.get();
+  // Symbol can be null in case of no rendering enabled
+  if ( !symbol )
+    return result;
+  const double pointSize = symbol->pointSize();
+
+  const double limitAngle = 2 * pointSize / screenSizePx * factory->mMap.fieldOfView();
```

Comment:
It would be good to add a comment here that we are currently using angle as the tolerance, therefore identifying points in a cone shape - not a cylinder (and that we may revisit that in future?)

Reply:
fixed